deblur 2021.09 (#3141)
qiita-spots/qiita: qiita_pet/support_files/doc/source/processingdata/deblur_2021.09.rst

deblur version 2021.09
======================

The deblur version 2021.09 addresses a bug with the fragment insertion parsing and
cache that ignored some fragments for getting an accurate placement in the tree. In
summary, in some occasions SEPP will return multiple fragments in a single entry; which
was unexpected by the qp-deblur plugin parser, which assumed only one entry - the
extra features will be seen as missing by the plugin and that information was
sent and stored in the cache provided by Qiita, then propagated to future studies and
meta-analyses.

This bug was resolved in this `pull request <https://github.com/qiita-spots/qp-deblur/pull/60>`__.

Sample counts implications
--------------------------

> At the time of writing Qiita had 978,052 16S deblured private or pubic samples.

Reviewer: unclear if you refer to the time of writing the bogus parser or this text

Author: this text, I can add more text to make it clearer ...

> In the figure below, we have at different trimming lengths how samples we will recover

Reviewer 2: ```suggestion
In the figure below, we have at different trimming lengths how many samples we recover
```

based on the minimum number of sequences per sample - this is an important consideration
as we normally need to remove samples below a given threshold for beta diversity

> calculations (via rarefactoin) or differential abundance testing.

Reviewer: this only applies, if rarefaction was based off faith PD, all other metrics should not have been affected, iff you did not additionally filter the deblur tables provided by Qiita down to those features contained in the insertion tree file (which comes together with the biom file in qiita)

Author: Good point, but this is true _only_ for the full/all table not for the reference-hit because the table needs to be filtered to match what's in the tree, which is automatically done via the meta-analysis construction in Qiita. Do you have any specific text suggestions to cover this ?

Author: I tried to add some more information about this at the end of the intro paragraph, please let me know if that works.

.. figure::  deblur2021.09_private_public.png
   :align:   center

A few conclusions from this plot:

- The maximum number of samples that we will recover are 6,771 at `Trimming (length: 150)`
  and min_seqs of 1,500; which represents a 0.7% increment in private and public samples.

> - At all Trimming lengths the curve tends to go up at up and then down based on min_seq,

Reviewer 2: ```suggestion
- At all Trimming lengths the curve tends to go up and then down based on min_seq,
```

  which is a common trend seen in rarefacion plots

Reaching out to affected study owners
-------------------------------------

As you saw in the previous section the effect of the missing fragments depends on the

> study, the trimming length and the minimal sequence count per sample selected. As a

Reviewer 2: ```suggestion
study, the trimming length and the minimum per sample sequence count. As a
```

general rule of thumb, as a first analytical pass for meta-analysis for 16S data, we use
5,000 sequences per sample and we prefer 150 base pair trimming. Thus, we directly
contacted all study owners that would recover more than 5% of the samples in their study

> (total 24).

Reviewer 2: Would it be OK to recommend to users that they reach out to qiita-help if they are concerned about how this affected them?

In the figure below, we have at different trimming lengths how samples we will recover
based on the minimum number of sequences per sample - this is an important consideration
as we normally need to remove samples below a given threshold for beta diversity
calculations (via rarefactoin) or differential abundance testing.

.. figure::  deblur2021.09_private_public.png
   :align:   center

A few conclusions from this plot:

- The maximum number of samples that we will recover are 6,771 at `Trimming (length: 150)`
  and min_seqs of 1,500; which represents a 0.7% increment in private and public samples.
- At all Trimming lengths the curve tends to go up and then down based on min_seq,
  which is a common trend seen in rarefacion plots

How perversive is this bug?
---------------------------

For a better assessment we are going to focus on only on the 150 bps trimming length. At
the time of writing of this documentation Qiita had 1,484 16S preparations that have a
150 bps deblur table. Of those:

- 96.6% of preparations had 0-10% of features lost
- 12.6% had 10-20% of the features lost
-  9.7% 30-40%

> -  6.9% 40-50%

Reviewer 3: I think it's important to give the full list, all the way to 90%.

Remember that the percentage reported above is inclusive at the next level, for example the studies with

> 40-50% lost are also accounted for at lower levels.

Reviewer 3: Maybe also include a comment that we did not find any strong patterns among the studies that were most greatly affected, whether they were from a specific sample type (according to empo category) or target 16S variable region.

deblur version 2021.09
======================

The deblur version 2021.09 addresses a bug with the fragment insertion parsing and
cache that ignored some fragments for getting an accurate placement in the tree. In
summary, in some occasions SEPP will return multiple fragments in a single entry; which
was unexpected by the qp-deblur plugin parser, which assumed only one entry - the

> extra features will be seen as missing by the plugin and that information was
> sent and stored in the cache provided by Qiita, then propagated to future studies and

Reviewer 4: ```suggestion
extra features will be seen as missing by the plugin and recorded in the cache as such. As these fragments were reported as missing in the cache, the effect was propagated to future studies and
```

meta-analyses.

This bug was resolved in this `pull request <https://github.com/qiita-spots/qp-deblur/pull/60>`__.

It is important to note that this bug only applies to the fragments inserted into the tree, which is
only part of the `deblur reference hit table`.

How do I know if my study processing had this bug?
----------------------------------------------------

The easiest is to check the table summary reported `Number rejected fragments`. If the number is
different between the qp-deblur v1.1.0 and qp-deblur v2021.09 then your study had this bug. To
see the table summary, you need to navigate to the processing graph, click on the
`deblur reference hit table` artifact and see the table summary.

Sample counts implications
--------------------------

At the time of writing of this documentation Qiita had 978,052 16S deblured private or pubic samples.
In the figure below, we have at different trimming lengths how samples we will recover
based on the minimum number of sequences per sample - this is an important consideration
as we normally need to remove samples below a given threshold for beta diversity

> calculations (via rarefactoin) or differential abundance testing.

Reviewer 4: ```suggestion
calculations (via rarefaction) or differential abundance testing.
```
